packages/smart-accounts/src/ma-v2/utils/deferredActions.ts

```diff
@@ -36,22 +44,23 @@ export type BuildDeferredActionDigestParams = {
  * @param {object} args The argument object containing the following:
  * @param {Hex} args.fullPreSignatureDeferredActionDigest The The data to append the signature and length to
  * @param {Hex} args.sig The signature to include in the digest
- * @param {Hex} args.signaturePrefix Prefix to append to the signature (defaults to "0x00" for EOA type)
  * @returns {Hex} The encoded digest to be prepended to the userOp signature
  */
 export const buildDeferredActionDigest = ({
   fullPreSignatureDeferredActionDigest,
   sig,
-  signaturePrefix = SignaturePrefix.EOA,
 }: BuildDeferredActionDigestParams): Hex => {
-  const prefixedSig = concatHex([signaturePrefix, sig]);
+  // 6492 sigs don't work here.
+  if (isErc6492Signature(sig)) {
```

Comment:
The `parseErc6492Signature` internally calls `isErc6492Signature` and if it isn't, it just returns the signature without modification. So we can remove the conditional here and always call `parseErc6492Signature`.